### PR TITLE
seal the plugins module and mark it as unstable

### PR DESCRIPTION
**Pull request by p-avital**
Since we might be planning changes to plugins (possibly removing the `Runtime` from their signatures), marking them as `unstable` makes sense.

### CODE REVIEW

File: zenoh/src/sample.rs [hunk 1]
```diff
@@ -33,6 +33,7 @@ pub enum Locality {
 #[derive(Clone, Copy, Debug, PartialEq, Eq)]
 pub(crate) enum Locality {
     SessionLocal,
+    #[allow(dead_code)]
```
[the remark this answers is not in the document]
Comment (p-avital) [reply zenoh/src/sample.rs:36]: Silencing a clippy lint (apparently the CI didn't catch it, but my rust-analyzer did). Another option would be to straight up remove that variant.

[the remark this answers is not in the document]
Comment (Mallets) [reply zenoh/src/sample.rs:36]: If that's not used, it should be removed. Opinions?

[the remark this answers is not in the document]
Comment (p-avital) [reply zenoh/src/sample.rs:36]: agreed

**Comment by Mallets** [comment]
Waiting to merge this PR. @p-avital could you please check with other zenoh plugins if this PR does not introduce any breaking change or some sister PR should follow?

**Comment by p-avital** [comment]
> Waiting to merge this PR. @p-avital could you please check with other zenoh plugins if this PR does not introduce any breaking change or some sister PR should follow?

`zenoh-plugin-dds` builds without changes
Sister PR opened `zenoh-plugin-webserver` eclipse-zenoh/zenoh-plugin-webserver#11

**Comment by p-avital** [comment]
Additional advantage of making plugins unstable: if any plugin wants to use unstable apis, zenohd must be compiled with `unstable` to have a chance of having compatible types, which means all other plugins must also be compiled with `unsable` regardless of their use of unstable APIs or not.

In fact, we may want to add a panic if zenohd detects that plugins have been requested despite having been built without `unstable`. Opinions?

**Comment by Mallets** [comment]
> Additional advantage of making plugins unstable: if any plugin wants to use unstable apis, zenohd must be compiled with `unstable` to have a chance of having compatible types, which means all other plugins must also be compiled with `unsable` regardless of their use of unstable APIs or not.
> 
> In fact, we may want to add a panic if zenohd detects that plugins have been requested despite having been built without `unstable`. Opinions?

I believe having an explicit message is always helpful for the user.

**Comment by JEnoch** [comment]
Well, considering that in 99% of cases `zenohd` will be built in the zenoh workspace that also contains `zenoh-ext` which depends on `zenoh` with `unstable` feature, is almost becomes a requirement for plugins and backends to also use zenoh's `unstable` feature. And that's automatic anyway for those also using `zenoh-ext`, such as `zenoh-plugin-dds`.

So yes, such detection and clear panic message would be nice.

**Review by Mallets — APPROVED** [review]

**Pull request merged**